### DISCUSSION TITLE
ERROR  Rollup error: Could not load /node_modules/pinia/dist/pinia.mjs/dist/pinia.mjs' in Nuxt 3

### DISCUSSION THREAD

**Discussion by Gyurmatag** [vuejs/pinia#1523] [Help and Questions]
Hi!

I have the latest Nuxt 3 within my project, version [3.0.0-rc.6-27660452.463c15e] and I've got the latest Pinia versions:
 "@pinia/nuxt": "^0.3.1"
"pinia": "^2.0.17",

and when I build I got this error
` ERROR  Rollup error: Could not load /Users/vargagyorgymark/Documents/Personal/Projektek/evgyuru/evgyuru-nuxt/node_modules/pinia/dist/pinia.mjs/dist/pinia.mjs (imported by .nuxt/dist/server/server.mjs): ENOTDIR: not a directory, open '/Users/vargagyorgymark/Documents/Personal/Projektek/evgyuru/evgyuru-nuxt/node_modules/pinia/dist/pinia.mjs/dist/pinia.mjs'`

What could be the problem?

Thank you very much!

**Comment by monkemedia** [comment]
I have the exact issue as you do. Have you managed to resolve it?

**Reply by Gyurmatag** [reply]
No... :( Still have the same issue with newer Nuxt 3 versions too... Maybe @posva can help us.

**Reply by monkemedia** [reply]
Yeah I am currently using the latest version of Nuxt. It's so frustrating as i am unable to test my site online

**Comment by posva** [comment]
It's as if the alias.pinia option was not taken into account correctly. Try adding one to the nuxt config that points to the absolute path of the pinia.mjs file. Something along the line of `path.resolve(__dirname, './node_modules/pinia/dist/pinia.mjs')`

**Reply by Gyurmatag** [reply]
Where and how should we put this line to the nuxt.config file?
Here is my nuxt.config.ts Where should I put the line that you wrote for us, @posva?

```
import { defineNuxtConfig } from "nuxt";

export default defineNuxtConfig({
  buildModules: ["@intlify/nuxt3"],
  modules: ["@nuxtjs/color-mode", "@pinia/nuxt"],
  intlify: {
    localeDir: "locales",
    vueI18n: {
      locale: "hu",
    },
  },
  css: ["@/assets/styles/main.css"],
  colorMode: {
    classSuffix: "",
  },
  build: {
    postcss: {
      postcssOptions: require("./postcss.config.js"),
    },
  },
  publicRuntimeConfig: {
    API_BASE: process.env.API_BASE,
  },
  meta: {
    link: [
      {
        href: "https://fonts.googleapis.com/css?family=Material+Icons|Material+Icons+Outlined",
        rel: "stylesheet",
      },
    ],
  },
});
```

Thank you very much!

**Reply by monkemedia** [reply]
Solved it. in your Nuxt config, add this:

```
import path from 'path'

export default defineNuxtConfig({
...
   resolve: {
      alias: {
        pinia: path.resolve(__dirname, './node_modules/pinia/dist/pinia.mjs'),
      }
    }
 ...
 })
 ```

**Reply by Gyurmatag** [reply]
Still not working with this added to nuxt.config. When I run nuxt build I've got the same error.
What sould I do @monkemedia , @posva ?

**Reply by monkemedia** [reply]
make sure you add this too:

```
const { __dirname } = createCommonJS(import.meta.url)
```

**Reply by posva** [reply]
Normally you don't need to add this so you might have something in your config that isn't necessary.

Check all the imports in your code from pinia and make sure you are not importing from a subfolder:

```js
import ... from 'pinia' // good
import ... from 'pinia/...' // bad
```

**Reply by Gyurmatag** [reply]
@monkemedia , @posva Still not working. I've added that code, that mankomedia wrote, but it's not working, same error message. I don't have any specific pinia import that is wrongly imported, I'm importing pinia this way
`import { defineStore } from "pinia";`

And also I've got this dependency for persisted state:
`"pinia-plugin-persistedstate": "^1.5.1",`

I have no idea what is wrong, but it worked with previous versions... Please help me!

**Comment by Alex--C** [comment]
Same issue here. Changing my `nuxt.config.ts` like suggested by @monkemedia does _not_ fix my issue.

Actually, removing `nuxt.options.alias.pinia = "pinia/dist/pinia.mjs";` in `node_modules/@pinia/nuxt/dist/module.mjs` fixes the issue during `yarn run build`, but then I get an error during runtime because pina cannot be imported.

**Reply by Gyurmatag** [reply]
Yeah. The issue still persist for me too. @posva can you please make a fix for that? Thank you very much!

**Comment by Gyurmatag** [comment]
@posva It is still not working on the latest nuxt release... Can you please make a fix for that? Or @Alex--C have you found any working workaround? It is extremely frustrating since I can't work on my project...

**Reply by Alex--C** [reply]
Amongst many other things I added this to my nuxt.config.ts and it's working now. Not 100% sure if that's what fixed it in the end but you can try it:

```
build: {
    transpile: [
      'pinia',
    ],
  },
```

**Reply by monkemedia** [reply]
I removed my changes and applied @Alex--C changes, and I can confirm it works :)

**Reply by Gyurmatag** [reply]
Yes, it's working for me too, thank you very much!
